ffi/src/dvc/dvc_pipe_proxy_message_queue.rs

```diff
@@ -0,0 +1,70 @@
+use ironrdp::svc::SvcMessage;
+use std::sync::mpsc;
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::error::ffi::IronRdpError;
+    use std::sync::mpsc;
+
+    use super::{DvcPipeProxyMessageInner, DvcPipeProxyMessageQueueInner};
+
+    #[diplomat::opaque]
+    pub struct DvcPipeProxyMessage(pub DvcPipeProxyMessageInner);
+
+    impl DvcPipeProxyMessage {
+        pub fn get_channel_id(&self) -> u32 {
+            self.0 .0
+        }
+    }
+
+    #[diplomat::opaque]
+    #[derive(Clone)]
+    pub struct DvcPipeProxyMessageSink(pub mpsc::SyncSender<DvcPipeProxyMessageInner>);
+
+    #[diplomat::opaque]
+    pub struct DvcPipeProxyMessageQueue(DvcPipeProxyMessageQueueInner);
+
+    impl DvcPipeProxyMessageQueue {
+        pub fn new(queue_size: u32) -> Box<DvcPipeProxyMessageQueue> {
+            let queue_size = usize::try_from(queue_size).expect("invalid dvc pipe proxy message queue size");
+
+            Box::new(DvcPipeProxyMessageQueue(DvcPipeProxyMessageQueueInner::new(queue_size)))
+        }
+
+        pub fn next_message(&self) -> Result<Option<Box<DvcPipeProxyMessage>>, Box<IronRdpError>> {
+            Ok(self.0.next_message().map(DvcPipeProxyMessage).map(Box::new))
+        }
+
+        pub fn next_message_blocking(&self) -> Result<Box<DvcPipeProxyMessage>, Box<IronRdpError>> {
+            let message = self.0.next_message_blocking().map(DvcPipeProxyMessage).map(Box::new)?;
+
+            Ok(message)
+        }
```

Comment:
suggestion: I suppose next_message will return None when no message is ready to be received? The conventional naming in Rust is to use `try_`: `try_next_message` for the non-blocking one, and `next_message` for the blocking one. May also be a good idea to add documentation.

Reply:
I see :thinking: 

I would be in favor of changing that, but can be a follow up!